packages/block-editor/src/components/block-drop-zone/index.js

```diff
@@ -116,17 +116,20 @@ class BlockDropZone extends Component {
 			return null;
 		}
 		const isAppender = index === undefined;
-
+		const dropZone = (
```

Comment:
Was thinking how to improve readability here to better communicate the difference. Some raw thoughts:

1) Do not use the `MediaUploadCheck` component but use the `hasUploadPermissions` prop:

```jsx
<DropZone
    className={ ... }
    onHTMLDrop={ this.onHTMLDrop }
    onDrop={ this.onDrop }
    onFilesDrop={ hasUploadPermissions ? this.onFilesDrop : null }
    />
```

2) Extract and rename the dropzone components so we return:

```jsx
return (
    <MediaUploadCheck
        fallback={ dropZoneWithoutFileDrop }>
        dropZoneWithFileDrop
    </MediaUploadCheck>
);
```

Reply:
Hi @nosolosw, I updated the code to follow your first suggestion.